lrauv_ignition_plugins/src/RangeBearingPlugin.cc

```diff
@@ -167,29 +169,45 @@ void RangeBearingPrivateData::PublishResponse(
   lrauv_ignition_plugins::msgs::LRAUVRangeBearingResponse resp;
   std::istringstream stream{msg.data()};
   resp.ParseFromIstream(&stream);
-  
+
   auto timeOfTx = transmissionTime[resp.req_id()];
   transmissionTime.erase(resp.req_id());
   auto duration = std::chrono::duration<double>(
     this->timeNow - timeOfTx - this->processingDelay);
   auto range = (this->speedOfSound * duration.count()) / 2;
 
+  // Get current pose
   auto poseOffset = ignition::math::Matrix4d(this->currentPose);
   auto otherVehiclesPos =
     ignition::math::Vector3d(
       resp.bearing().x(), resp.bearing().y(), resp.bearing().z());
-  auto groundTruthPose = poseOffset.Inverse() * otherVehiclesPos;
-
-  auto theta = acos(groundTruthPose.Z() / groundTruthPose.Length());
-  auto phi = atan2(groundTruthPose.Y(), groundTruthPose.X());
+  // Transform pose of other vehicle to local frame
+  auto poseInLocalFrame = poseOffset.Inverse() * otherVehiclesPos;
+
+  igndbg << "Current pose " << poseOffset.Pose().Pos() << "R: " << poseOffset.Pose().Rot().Euler() << "\n";
+  igndbg << "Target pose (global frame)" << otherVehiclesPos << "\n";
+  igndbg << "Target pose (local frame)" << poseInLocalFrame << "\n";
+
+  // Elevation is given as a function of angle from XY plane of the vehicle 
+  // with positive facing down.
+  auto negativeZAxis = ignition::math::Vector3d(0, 0, -1);
+  auto elev =
+    asin(negativeZAxis.Dot(poseInLocalFrame) / poseInLocalFrame.Length());
+
+  // Azimuth is given by the angle from the X axis in vehicle frame.
+  auto xyProj = ignition::math::Vector2d(poseInLocalFrame.X(),
+    poseInLocalFrame.Y());
+  // TODO(arjo): This minus sign shouldn't be necessary. 
+  auto azimuth = (xyProj.Length() < 0.001) ? 0 : -atan2(xyProj.Y(), xyProj.X());
+  igndbg << "theta " << elev << " phi " << azimuth << "\n";
```

Comment:
@arjo129 replace theta/phi with vehicle frame elevation/azimuth. For reference, the debug output in the LRAUV app looks like this:
```
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): slant range: 51.063232 m
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): rotation FSK->NED: [[-0.115604,0.992562,0.038176],[-0.952876,-0.121670,0.277892],[0.280470,-0.004252,0.959853]]
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): direction FSK: [0.229616,0.897985,-0.375366]
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): direction NED: [0.850431,-0.432364,-0.299714]
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): rel pos FSK:   [11.724925,45.854018,-19.167385]
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): rel pos NED:   [43.425751,-22.077918,-15.304351]
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): FSK: azumith=75.656735 deg, elevation=-22.046916 deg.
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): NED: bearing=-26.949063 deg, inclination=-17.440408 deg.
2022-03-28T23:38:33.930Z,1648510713.930 [circle_acoustic_contact:B.](INFO): NED: lat=36.803405 deg, lon=-121.827208 deg, depth=-0.423813 m.
```